numpy/core/src/umath/ufunc_object.c

```diff
@@ -1316,17 +1304,20 @@ iterator_loop(PyUFuncObject *ufunc,
         return -1;
     }
 
-    /* Copy any allocated outputs */
-    op_it = NpyIter_GetOperandArray(iter);
-    for (i = 0; i < nout; ++i) {
-        if (op[nin+i] == NULL) {
-            op[nin+i] = op_it[nin+i];
-            Py_INCREF(op[nin+i]);
+    NPY_UF_DBG_PRINT("Made iterator\n");
+
+    /* Call the __array_prepare__ functions where necessary */
```

Comment:
Adopted (the comment exists twice, but the second point is deleted in the followup).

Yeah, now that the full change is ready, it is surprising (to silly me) how much I can probably split out to keep the big change smaller or at least simpler.  (Not sure how many more small(ish) PRs it will be/are possible.)

Reply:
Great! These small PRs are much easier to review, so while it will be a pain for you, I think it is well worth it. For the big one, one can then focus on the real changes!